src/starkware_utils/commitment_tree/binary_fact_tree.rs

```diff
@@ -11,6 +12,14 @@ pub trait Leaf: Clone {}
 
 pub type BinaryFactDict = HashMap<BigUint, Vec<BigUint>>;
 
+/// Converts a BinaryFactDict from maps of BigUints to Felt252s.
+pub fn binary_fact_dict_to_felts(binary_fact_dict: BinaryFactDict) -> HashMap<Felt252, Vec<Felt252>> {
```

Comment:
The implementation of the Patricia tree stuff. I decided to avoid using felts in the implementation whenever possible as they are not needed. This requires a conversion function here and there to interact with Cairo code.

For this particular structure we could decide to use Felts from the start. I don't think it has any impact in terms of performance at this stage so I'd rather keep the Patricia tree code as is and keep this optimization for later.

Reply:
Felts are optimized for field operations(https://en.wikipedia.org/wiki/Montgomery_modular_multiplication), are you sure it makes sense? 
And conversion are not for free. I suppose in the future we will get felts directly from the sequencer, so two conversions, to/from BigUints will be necessary.

Reply:
The container conversions and the heap allocations those require is also expensive.

That said, I doubt any of this is significant when compared to the overall cost of running the prover...

Reply:
For what calculations those BigUints are used for?

Reply:
Mostly descent map stuff.